pyup/requirements.py

```diff
@@ -232,6 +232,18 @@ def __init__(self, name, specs, line, lineno, extras, file_type):
         self._is_insecure = None
         self._changelog = None
 
+        if len(self.specs._specs) == 1 and next(iter(self.specs._specs))._spec[0] == "~=":
```

Comment:
Touching these private attributes drives me crazy, but we have been doing this for some time already. :man_shrugging: 

Nice job keeping the same code style. :+1: